,dep,version,filepath,bloated,cve,affected versions,advisory
0,skl2onnx,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements.txt,False,no one,no one,no one
1,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements.txt,False,CVE-2017-12852,<1.13.2,"Numpy 1.13.2 includes a fix for CVE-2017-12852: The numpy.pad function in Numpy 1.13.1 and older versions is missing input validation. An empty list or ndarray will stick into an infinite loop, which can allow attackers to cause a DoS attack.
https://github.com/numpy/numpy/issues/9560#issuecomment-322395292"
2,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements.txt,False,CVE-2019-6446,<1.16.3,"Numpy 1.16.3 includes a fix for CVE-2019-6446: It uses the pickle Python module unsafely, which allows remote attackers to execute arbitrary code via a crafted serialized object, as demonstrated by a numpy.load call.
https://github.com/numpy/numpy/commit/89b688732b37616c9d26623f81aaee1703c30ffb"
3,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements.txt,False,CVE-2021-33430,<1.21.0rc1,"Numpy 1.21.0rc1 includes a fix for CVE-2021-33430: A Buffer Overflow vulnerability in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service. 
NOTE: The vendor does not agree this is a vulnerability. In (very limited) circumstances a user may be able provoke the buffer overflow, the user is most likely already privileged to at least provoke denial of service by exhausting memory. Triggering this further requires the use of uncommon API (complicated structured dtypes), which is very unlikely to be available to an unprivileged user.
https://github.com/numpy/numpy/issues/18939"
4,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements.txt,False,CVE-2021-34141,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-34141: An incomplete string comparison in the numpy.core component in NumPy before 1.22.0 allows attackers to trigger slightly incorrect copying by constructing specific string objects. 
NOTE: the vendor states that this reported code behavior is ""completely harmless.""
https://github.com/numpy/numpy/issues/18993"
5,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements.txt,False,CVE-2021-41496,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-41496: Buffer overflow in the array_from_pyobj function of fortranobject.c, which allows attackers to conduct a Denial of Service attacks by carefully constructing an array with negative values. 
NOTE: The vendor does not agree this is a vulnerability; the negative dimensions can only be created by an already privileged user (or internally).
https://github.com/numpy/numpy/issues/19000"
6,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements.txt,False,CVE-2021-41495,<1.22.2,"Numpy 1.22.2  includes a fix for CVE-2021-41495: Null Pointer Dereference vulnerability exists in numpy.sort in NumPy in the PyArray_DescrNew function due to missing return-value validation, which allows attackers to conduct DoS attacks by repetitively creating sort arrays. 
NOTE: While correct that validation is missing, an error can only occur due to an exhaustion of memory. If the user can exhaust memory, they are already privileged. Further, it should be practically impossible to construct an attack which can target the memory exhaustion to occur at exactly this place.
https://github.com/numpy/numpy/issues/19038"
7,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements.txt,False,CVE-2014-1859,<1.8.1,"Numpy 1.8.1 includes a fix for CVE-2014-1859: (1) core/tests/test_memmap.py, (2) core/tests/test_multiarray.py, (3) f2py/f2py2e.py, and (4) lib/tests/test_io.py in NumPy before 1.8.1 allow local users to write to arbitrary files via a symlink attack on a temporary file.
https://github.com/numpy/numpy/pull/4262"
8,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements.txt,False,CVE-2014-1858,<1.8.1,"Numpy 1.8.1 includes a fix for CVE-2014-1858: __init__.py in f2py in NumPy before 1.8.1 allows local users to write to arbitrary files via a symlink attack on a temporary file.
https://github.com/numpy/numpy/commit/0bb46c1448b0d3f5453d5182a17ea7ac5854ee15"
9,onnx,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements.txt,False,CVE-2022-25882,<1.13.0,Onnx 1.13.0 includes a fix for CVE-2022-25882: Versions of the package onnx before 1.13.0 are vulnerable to Directory Traversal as the external_data field of the tensor proto can have a path to the file which is outside the model current directory or user-provided directory.
10,lightgbm,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
11,xgboost,==1.7.5,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
12,catboost,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2017-15288,<0.26,"Catboost 0.26 updates version of 'scala' to v2.11.12 for security reasons.
https://github.com/catboost/catboost/issues/1632"
13,scipy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2013-4251,">=0,<0.12.1",The scipy.weave component in SciPy before 0.12.1 creates insecure temporary directories.
14,openpyxl,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2017-5992,">=0,<2.4.1","Openpyxl 2.4.1 resolves external entities by default, which allows remote attackers to conduct XXE attacks via a crafted .xlsx document."
15,mleap,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
16,scikit-learn,>=1.2.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2020-13092,">=0,<0.23.1","** DISPUTED ** scikit-learn (aka sklearn) through 0.23.0 can unserialize and execute commands from an untrusted file that is passed to the joblib.load() function, if __reduce__ makes an os.system call. NOTE: third parties dispute this issue because the joblib.load() function is documented as unsafe and it is the user's responsibility to use the function in a secure manner."
17,scikit-learn,>=1.2.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2020-28975,">=0,<0.24.dev0","** DISPUTED ** svm_predict_values in svm.cpp in Libsvm v324, as used in scikit-learn 0.23.2 and other products, allows attackers to cause a denial of service (segmentation fault) via a crafted model SVM (introduced via pickle, json, or any other model permanence standard) with a large value in the _n_support array. NOTE: the scikit-learn vendor's position is that the behavior can only occur if the library's API is violated by an application that changes a private attribute."
18,pyspark,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2019-10099,<2.3.3,"Prior to Spark 2.3.3, in certain situations Spark would write user data to local disk unencrypted, even if spark.io.encryption.enabled=true. This includes cached blocks that are fetched to disk (controlled by spark.maxRemoteBlockSizeFetchToMem); in SparkR, using parallelize; in Pyspark, using broadcast and parallelize; and use of python udfs."
19,pyspark,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2020-9480,">=0,<2.4.6","In Apache Spark 2.4.5 and earlier, a standalone resource manager's master may be configured to require authentication (spark.authenticate) via a shared secret. When enabled, however, a specially-crafted RPC to the master can succeed in starting an application's resources on the Spark cluster, even without the shared key. This can be leveraged to execute shell commands on the host machine. This does not affect Spark clusters using other resource managers (YARN, Mesos, etc)."
20,pyspark,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2021-38296,">=0,<3.1.3","Apache Spark supports end-to-end encryption of RPC connections via ""spark.authenticate"" and ""spark.network.crypto.enabled"". In versions 3.1.2 and earlier, it uses a bespoke mutual authentication protocol that allows for full encryption key recovery. After an initial interactive attack, this would allow someone to decrypt plaintext traffic offline. Note that this does not affect security mechanisms controlled by ""spark.authenticate.enableSaslEncryption"", ""spark.io.encryption.enabled"", ""spark.ssl"", ""spark.ui.strictTransportSecurity"". Update to Apache Spark 3.1.3 or later"
21,pyspark,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2022-31777,">=0,<3.2.2","A stored cross-site scripting (XSS) vulnerability in Apache Spark 3.2.1 and earlier, and 3.3.0, allows remote attackers to execute arbitrary JavaScript in the web browser of a user, by including a malicious payload into the logs which would be returned in logs rendered in the UI."
22,pyspark,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2018-1334,">=2.2.0,<2.2.2,>=0,<2.1.3","In Apache Spark 1.0.0 to 2.1.2, 2.2.0 to 2.2.1, and 2.3.0, when using PySpark or SparkR, it's possible for a different local user to connect to the Spark application and impersonate the user running the Spark application."
23,pyspark,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2018-11760,">=2.3.0,<2.3.2,>=1.0.2,<2.2.3","When using PySpark , it's possible for a different local user to connect to the Spark application and impersonate the user running the Spark application. This affects versions 1.x, 2.0.x, 2.1.x, 2.2.0 to 2.2.2, and 2.3.0 to 2.3.1."
24,pyspark,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2022-33891,">=3.2.0,<3.2.2,>=3.1.1,<3.1.3,<=3.0.3","Pyspark 3.1.3 and 3.2.2 includes a fix for CVE-2022-33891: The Apache Spark UI offers the possibility to enable ACLs via the configuration option spark.acls.enable. With an authentication filter, this checks whether a user has access permissions to view or modify the application. If ACLs are enabled, a code path in HttpSecurityFilter can allow someone to perform impersonation by providing an arbitrary user name. A malicious user might then be able to reach a permission check function that will ultimately build a Unix shell command based on their input, and execute it. This will result in arbitrary shell command execution as the user Spark is currently running as. This affects Apache Spark versions 3.0.3 and earlier, versions 3.1.1 to 3.1.2, and versions 3.2.0 to 3.2.1."
25,pytest-cov,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
26,psutil,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2019-18874,<=5.6.5,psutil (aka python-psutil) through 5.6.5 can have a double free. This occurs because of refcount mishandling within a while or for loop that converts system data into a Python object. See CVE-2019-18874.
27,flake8,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
28,libsvm,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
29,flatbuffers,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
30,pandas,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2020-13091,">=0,<1.0.4","** DISPUTED ** pandas through 1.0.3 can unserialize and execute commands from an untrusted file that is passed to the read_pickle() function, if __reduce__ makes an os.system call. NOTE: third parties dispute this issue because the read_pickle() function is documented as unsafe and it is the user's responsibility to use the function in a secure manner."
31,pytest-spark,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
32,pytest,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
33,dill,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
34,cython,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,no one,no one,no one
35,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2017-12852,<1.13.2,"Numpy 1.13.2 includes a fix for CVE-2017-12852: The numpy.pad function in Numpy 1.13.1 and older versions is missing input validation. An empty list or ndarray will stick into an infinite loop, which can allow attackers to cause a DoS attack.
https://github.com/numpy/numpy/issues/9560#issuecomment-322395292"
36,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2019-6446,<1.16.3,"Numpy 1.16.3 includes a fix for CVE-2019-6446: It uses the pickle Python module unsafely, which allows remote attackers to execute arbitrary code via a crafted serialized object, as demonstrated by a numpy.load call.
https://github.com/numpy/numpy/commit/89b688732b37616c9d26623f81aaee1703c30ffb"
37,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2021-33430,<1.21.0rc1,"Numpy 1.21.0rc1 includes a fix for CVE-2021-33430: A Buffer Overflow vulnerability in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service. 
NOTE: The vendor does not agree this is a vulnerability. In (very limited) circumstances a user may be able provoke the buffer overflow, the user is most likely already privileged to at least provoke denial of service by exhausting memory. Triggering this further requires the use of uncommon API (complicated structured dtypes), which is very unlikely to be available to an unprivileged user.
https://github.com/numpy/numpy/issues/18939"
38,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2021-34141,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-34141: An incomplete string comparison in the numpy.core component in NumPy before 1.22.0 allows attackers to trigger slightly incorrect copying by constructing specific string objects. 
NOTE: the vendor states that this reported code behavior is ""completely harmless.""
https://github.com/numpy/numpy/issues/18993"
39,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2021-41496,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-41496: Buffer overflow in the array_from_pyobj function of fortranobject.c, which allows attackers to conduct a Denial of Service attacks by carefully constructing an array with negative values. 
NOTE: The vendor does not agree this is a vulnerability; the negative dimensions can only be created by an already privileged user (or internally).
https://github.com/numpy/numpy/issues/19000"
40,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2021-41495,<1.22.2,"Numpy 1.22.2  includes a fix for CVE-2021-41495: Null Pointer Dereference vulnerability exists in numpy.sort in NumPy in the PyArray_DescrNew function due to missing return-value validation, which allows attackers to conduct DoS attacks by repetitively creating sort arrays. 
NOTE: While correct that validation is missing, an error can only occur due to an exhaustion of memory. If the user can exhaust memory, they are already privileged. Further, it should be practically impossible to construct an attack which can target the memory exhaustion to occur at exactly this place.
https://github.com/numpy/numpy/issues/19038"
41,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2014-1859,<1.8.1,"Numpy 1.8.1 includes a fix for CVE-2014-1859: (1) core/tests/test_memmap.py, (2) core/tests/test_multiarray.py, (3) f2py/f2py2e.py, and (4) lib/tests/test_io.py in NumPy before 1.8.1 allow local users to write to arbitrary files via a symlink attack on a temporary file.
https://github.com/numpy/numpy/pull/4262"
42,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2014-1858,<1.8.1,"Numpy 1.8.1 includes a fix for CVE-2014-1858: __init__.py in f2py in NumPy before 1.8.1 allows local users to write to arbitrary files via a symlink attack on a temporary file.
https://github.com/numpy/numpy/commit/0bb46c1448b0d3f5453d5182a17ea7ac5854ee15"
43,wheel,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/onnxmltools/requirements-dev.txt,False,CVE-2022-40898,<0.38.1,"Wheel 0.38.1 includes a fix for CVE-2022-40898: An issue discovered in Python Packaging Authority (PyPA) Wheel 0.37.1 and earlier allows remote attackers to cause a denial of service via attacker controlled input to wheel cli.
https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages"
